File: javascript/largest-series-product/largest-series-product.js
export const largestProduct = (inputNumber, numberOfDigits) => {
    if (/\D/.test(inputNumber)) {
        throw new Error('Digits input must only contain digits');
    }

    if (numberOfDigits <= 0) {
        throw new Error('Span must be greater than zero');
    }

    if (inputNumber.length < numberOfDigits) {
        throw new Error('Span must be smaller than string length');

    }

    return [...Array(inputNumber.length - numberOfDigits + 1).keys()]
        .map(pos => product(inputNumber.slice(pos, pos + numberOfDigits)))
        .reduce((acc, current) => Math.max(acc, current));
};

const product = (inputNumber) => 
    [...inputNumber].map(Number).reduce((a, b) => a * b, 1);

File: python/largest-series-product/largest_series_product.py
def largest_product(series, size):
    
    if size > len(series):
        raise ValueError("span must be smaller than string length")
    
    if size < 0:
        raise ValueError("span must not be negative")
    
    if not series.isnumeric():
        raise ValueError("digits input must only contain digits")

    
    max_value = 0

    for i in range(len(series) - size + 1):
        num = series[i:i + size]
        max_value = max(max_value, str_prod(num))

    return max_value


def str_prod(number):
    prod = 1

    for num in number:
        prod = prod * int(num)

    return prod

